src/main/java/net/tiagofar78/prisonescape/Events.java

```diff
@@ -119,38 +120,50 @@ public void playerCloseInventory(InventoryCloseEvent e) {
         game.playerCloseMenu(e.getPlayer().getName());
     }
 
-    @SuppressWarnings("deprecation")
     @EventHandler
     public void playerClickInventory(InventoryClickEvent e) {
         PrisonEscapeGame game = GameManager.getGame();
         if (game == null) {
             return;
         }
 
-        if (e.getInventory() == null) {
+        if (e.getClickedInventory() == null) {
             return;
         }
 
+        boolean isPlayerInv = false;
         if (e.getClickedInventory().getType() == InventoryType.PLAYER) {
-            return;
+            Inventory topInv = e.getView().getTopInventory();
+            if (topInv == null) {
+                e.setCancelled(true);
+                return;
+            }
+
+            isPlayerInv = true;
         }
 
+        ItemStack cursor = e.getCursor();
+        ItemStack current = e.getCurrentItem();
+
         PrisonEscapeItem item = BukkitItems.convertToPrisonEscapeItem(e.getCursor());
-        ClickReturnAction returnAction = game.playerClickMenu(e.getWhoClicked().getName(), e.getSlot(), item);
+        ClickReturnAction returnAction = game.playerClickMenu(
+                e.getWhoClicked().getName(),
```

Comment:
I've noticed that you frequently use e.getWhoClicked() in this PR. I suggest introducing a variable Player player = e.getWhoClicked() and utilizing it instead. While it's not a dealbreaker to keep it as is, I think it would be more readable. 

Up to you

Reply:
you are right, I will do it.